inputs/org.osbuild.files

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python3
+"""
+Inputs for individual files
+
+Provides all the files, named via their content hash, specified
+via `references` in a new directory.
+
+The returned data in `refs` is a dictionary where the keys are
+the content hash of a file and the values dictionaries with
+metadata for it. The input itself currently does not set any
+metadata itself, but will forward any metadata set via the
+`metadata` property. Keys in that must start with a prefix,
+like `rpm.` to avoid namespace clashes. This is enforced via
+schema validation.
+"""
+
+
+import json
+import sys
+import subprocess
+
+from osbuild.objectstore import StoreClient
+
+
+SCHEMA = r"""
+"additionalProperties": false,
+"required": ["origin", "references"],
+"properties": {
+  "origin": {
+    "description": "The origin of the input (must be 'org.osbuild.source')",
+    "type": "string",
+    "enum": ["org.osbuild.source"]
+  },
+  "references": {
```

Comment:
This is `refs` in the code below.

Reply:
Yeah, this was on purpose, i.e. short in code, where one has to type it, longer and more explicit in the manifest where it is being generated. But I am happy to change this to `refs` as well, if you have a strong objection. NB: now with the different formats there is not such strong correspondence between format and code anymore anyway.

Reply:
But it looks for `args["refs"]` below, which is the key that's being described in this schema, no?

Or am I totally confused?

Reply:
It is indeed a bit confusing, it comes from `inputs.py`, [here](https://github.com/osbuild/osbuild/pull/573/files#diff-2968c49f04f12d723142285ebf704cf74f6d66d6198d8771ae585f40c7638cb9R64), i.e. it the communication from osbuild to the input.

Reply:
ahhh thank you :heart: